### ISSUE TITLE
Swagger 2 :  Getting "type": "ref" when using @RequestPart

### ISSUE THREAD

**Issue by jrishabh198** [springfox/springfox#2097]
My controller
```java
@RequestMapping(value = "worker", method = RequestMethod.POST, consumes = {"multipart/form-data"})
    @ApiOperation(value = "create a worker")
    public void create(@RequestPart(name = "worker") Worker worker,
                       @RequestPart(name = "file")MultipartFile file)
    {
```

Worker 
```java
@ApiModel
public class Worker {
    @ApiModelProperty(value = "name")
    @NotNull
    @Size(min=1,max=50, message="Name must not exceed 50 characters")
    String name;
```

Generated swagger spec
```json
"/api/v1/worker": {
            "post": {
                "tags": [
                    "worker-controller"
                ],
                "summary": "create a worker",
                "operationId": "createUsingPOST",
                "consumes": [
                    "multipart/form-data"
                ],
                "produces": [
                    "*/*"
                ],
                "parameters": [
                    {
                        "name": "worker",
                        "in": "formData",
                        "description": "worker",
                        "required": true,
                        **"type": "ref"**
                    },
                    {
                        "name": "file",
                        "in": "formData",
                        "description": "file",
                        "required": true,
                        "type": "file"
                    }
                ],

...

"Worker": {
            "type": "object",
            "properties": {
                "name": {
                    "type": "string",
                    "description": "name"
                }
            }
        }
```
The parameter 'worker' is not getting mapped to the 'Worker' model

**Comment by dilipkrish** [comment]
Are you using v 2.7.0?

**Comment by jrishabh198** [comment]
Yes, using v 2.7.0

**Comment by dilipkrish** [comment]
This is fixed via #1965
